com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -979,6 +964,38 @@ define([
                     $scope.error = errorMessage;
                 });
             }
+
+          /**
+           * Trigger validation on JobRole Dates + attach error callback
+           * @param {object} data - The dates to validate
+           * @param {object} errors - The error recipients
+           * @returns {boolean|string}
+           */
+            function validateDates(data, errors) {
+              var errorsCount = 0;
+
+              DateValidation.setErrorCallback(function (error, field) {
+                  errorsCount++;
+                  if (field.indexOf('start_date') > -1) {
+                      errors.start.push(error);
+                  }
+                  if (field.indexOf('end_date') > -1) {
+                      errors.end.push(error);
+                  }
+              });
+              DateValidation.validate(data.start, data.end, data.contractStart, data.contractEnd);
+
+              return errorsCount > 0 ? 'Error' : true;
```

Comment:
Why do you return 'Error'? In case there are no errors, the function returns `true`, but it there are errors, the function returns a non-empty string, which also coerces to `true`. Can this function return `false`?